File: python/armstrong-numbers/armstrong_numbers.py
import math


def is_armstrong_number(number):
    numDigits = len(str(number))
    armNumCandidate = 0
    for i in range(numDigits):
        armNumCandidate += math.pow(int(str(number)[i]), numDigits)
    return armNumCandidate == number

File: go/armstrong-numbers/armstrong_numbers.go
package armstrong

import (
	"math"
	"strconv"
)

func IsNumber(n int) bool {
	numToString := strconv.Itoa(n)
	powerMultiplier := len(numToString)
	sumDigitsToPower := 0
	for _, digit := range numToString {
		digitInt, _ := strconv.Atoi(string(digit))
		sumDigitsToPower += int(math.Pow(float64(digitInt), float64(powerMultiplier)))
	}
	return sumDigitsToPower == n
}
